### PR TITLE
Toggle all feature for all variables and all deployment actions

### PR DESCRIPTION
**Pull request by taliesins**
Provide a way for people to toggle all variables actions
Provide a way for people to toggle all deployment actions
When a component has been deployed successfully, then marked it as skipped in the GUI. This will mean that when fixing and retrying it will only redeploy the failed ones.

### CODE REVIEW

File: src/OctopusPuppet.Gui/ViewModels/DeploymentPlannerViewModel.cs [hunk 1]
```diff
@@ -657,6 +657,157 @@ public void LoadEnvironmentDeploymentJson()
...
+                .ToList();
+
+            foreach (var successfullyDeployedComponent in successfullyDeployedComponents)
+            {
+                successfullyDeployedComponent.Vertex.DeploymentAction = PlanAction.Skip;
+            }
+
+            foreach (var componentToDeploy in componentsToDeploy)
+            {
+                componentToDeploy.Vertex.DeploymentAction = PlanAction.Skip;
```
Comment (RavindraSolanki) [inline src/OctopusPuppet.Gui/ViewModels/DeploymentPlannerViewModel.cs]: I'm not sure I understand why we're setting the components to deploy to be skipped.

Comment (taliesins) [inline src/OctopusPuppet.Gui/ViewModels/DeploymentPlannerViewModel.cs]: When we have successfully deployed a component, we don't want to have to deploy this component again. If a another component breaks and we resume deploying we want it to only deploy the components that have not been deployed yet or broke during deployment.

Comment (RavindraSolanki) [inline src/OctopusPuppet.Gui/ViewModels/DeploymentPlannerViewModel.cs]: I thought this was the the one above 'successfullyDeployedComponent'. And was expecting 'componentToDeploy' to be set as to deploy.

Comment (RavindraSolanki) [inline src/OctopusPuppet.Gui/ViewModels/DeploymentPlannerViewModel.cs]: Maybe it's just the way it's named. If not relevant ignore my comment. For the rest from what I understand it seems to be good

**Pull request merged**